tests/bluetooth/audio/ascs/src/main.c

```diff
@@ -0,0 +1,767 @@
+{
+	struct ascs_ase_control_test_suite_fixture *fixture = f;
+
+	ARG_UNUSED(fixture);
+
+	bt_conn_ref_fake.custom_fake = conn_ref_custom_fake;
+	bt_conn_get_info_fake.custom_fake = conn_get_info_custom_fake;
+	bt_gatt_foreach_attr_type_fake.custom_fake = gatt_foreach_attr_type_custom_fake;
+	bt_pacs_cap_foreach_fake.custom_fake = pacs_cap_foreach_custom_fake;
+
+	bap_unicast_server_cb_config_fake.return_val = 0;
+	bap_unicast_server_cb_reconfig_fake.return_val = 0;
+	bap_unicast_server_cb_qos_fake.return_val = 0;
+	bap_unicast_server_cb_enable_fake.return_val = 0;
+	bap_unicast_server_cb_start_fake.return_val = 0;
+	bap_unicast_server_cb_metadata_fake.return_val = 0;
+	bap_unicast_server_cb_disable_fake.return_val = 0;
+	bap_unicast_server_cb_stop_fake.return_val = 0;
+	bap_unicast_server_cb_release_fake.return_val = 0;
+}
+
+static void ascs_ase_control_test_suite_after(void *f)
+{
+	ascs_cleanup();
+}
+
+static void ascs_ase_control_test_suite_teardown(void *f)
+{
+	struct ascs_ase_control_test_suite_fixture *fixture = f;
+
+	bt_bap_unicast_server_unregister_cb(&unicast_server_cb);
+
+	free(fixture);
+}
+
+ZTEST_SUITE(ascs_ase_control_test_suite, NULL, ascs_ase_control_test_suite_setup,
+	    ascs_ase_control_test_suite_before, ascs_ase_control_test_suite_after,
+	    ascs_ase_control_test_suite_teardown);
+
+ZTEST_F(ascs_ase_control_test_suite, test_sink_ase_read_state_idle)
+{
+	uint8_t data_expected[] = {
+		0x01,   /* ASE_ID */
+		0x00,   /* ASE_State = Idle */
+	};
+	ssize_t ret;
+
+	ret = fixture->ase_snk_1->read(&fixture->conn, fixture->ase_snk_1, NULL, 0, 0);
+	zassert_false(ret < 0, "attr->read returned unexpected (err 0x%02x)",
+		      BT_GATT_ERR(ret));
+
+	expect_single_call_bt_gatt_attr_read(&fixture->conn, fixture->ase_snk_1, NULL, 0, 0,
+					     data_expected, ARRAY_SIZE(data_expected));
+}
+
+ZTEST_F(ascs_ase_control_test_suite, test_sink_ase_control_operation_zero_length_write)
+{
+	uint8_t buf[] = {};
+	ssize_t ret;
+
+	ret = fixture->ase_cp->write(&fixture->conn, fixture->ase_cp, (void *)buf, 0, 0, 0);
+	zassert_true(ret < 0, "ase_cp_attr->write returned unexpected (err 0x%02x)",
+		     BT_GATT_ERR(ret));
+}
+
+ZTEST_F(ascs_ase_control_test_suite, test_sink_ase_invalid_opcode)
+{
+	uint8_t buf[] = {
+		0x00,   /* Opcode */
+		0x01,   /* Number_of_ASEs */
+		0x01,   /* ASE_ID[0] */
+	};
+	uint8_t data_expected[] = {
+		0x00,   /* Opcode */
+		0xFF,   /* Number_of_ASEs */
+		0x00,   /* ASE_ID[0] */
+		0x01,   /* Response_Code[0] = Unsupported Opcode */
+		0x00,   /* Reason[0] */
+	};
+
+	bt_conn_index_fake.return_val = 0;
+
+	fixture->ase_cp->write(&fixture->conn, fixture->ase_cp, (void *)buf, sizeof(buf), 0, 0);
+
+	expect_single_call_bt_gatt_notify_cb(&fixture->conn, BT_UUID_ASCS_ASE_CP, fixture->ase_cp,
+					     data_expected, sizeof(data_expected));
+}
+
+ZTEST_F(ascs_ase_control_test_suite, test_sink_ase_unsupported_opcode)
+{
+	uint8_t buf[] = {
+		0xFE,   /* Opcode */
```

Comment:
How is `0xFE` unsuported, but `0x00` invalid? Aren't they both invalid?

Reply:
To explain myself,
`0x00` is not mentioned, meaning it's invalid
`0x09-0xFF` are RFU
but I agree, this can be put into one test case

Reply:
> 0x00 is not mentioned, meaning it's invalid

Is that true? The spec defines RFU and Prohibited, but I don't see where it states that a undefined value indicates that it is invalid. 

Does the test spec say anything here?

Reply:
![image](https://user-images.githubusercontent.com/10126877/225664464-4376797e-fdc8-4e3a-aa18-27ae8dbe91fc.png)
![image](https://user-images.githubusercontent.com/10126877/225664535-0e0b082f-b10f-4655-93db-6f7ba541b362.png)
there is no mention about 0x00